File: src-tauri/src/commands/platform.rs
use tauri::AppHandle;

use crate::{enums::error::Error, models::platform::Platform, services::platform::PlatformService};

#[tauri::command]
pub async fn command_get_platforms(app_handle: AppHandle) -> Result<Vec<Platform>, Error> {
    PlatformService::get_platforms(&app_handle).await
}

File: src/utils/http/platform.ts
import {Platform} from '@/models/platform'
import {TauriCommandKey, TauriCommandPayload, tauriInvoke} from '.'

export const getPlatforms = async (): Promise<TauriCommandPayload<Platform>> => {
	return tauriInvoke(TauriCommandKey.GET_PLATFORMS)
}
